src/radical/pilot/raptor/worker.py

```diff
@@ -93,9 +93,10 @@ def __init__(self, cfg, manager, rank):
         self.register_mode(TASK_SHELL,    self._dispatch_shell)
 
         # prepare base env dict used for all tasks
+        # NOTE: raptor tasks run in the same environment as the raptor worker
         self._task_env = dict()
         for k,v in os.environ.items():
-            if k.startswith('RP_'):
+            if not k.startswith('RP_'):
                 self._task_env[k] = v
```

Comment:
The most obvious error in issue #2889 is the TypeError from trying to set environment variables to non-string values. Where should the normalization be added?

Reply:
Will add, thanks.